drivers/place/attendee_scanner.cr

```diff
@@ -0,0 +1,88 @@
+require "placeos-driver"
+require "json"
+
+class Place::AttendeeScanner < PlaceOS::Driver
+  descriptive_name "PlaceOS Room Events"
+  generic_name :AttendeeScanner
+
+  accessor staff_api : StaffAPI_1
+
+  default_settings({
+    internal_domains: [] of String,
+  })
+
+  getter systems : Hash(String, Array(String)) { get_systems_list.not_nil! }
+  getter internal_domains : Array(String) = [] of String
+
+  def on_update
+    @internal_domains = setting(Array(String), :internal_domains)
+  end
+
+  # Grabs the list of systems in the building
+  def get_systems_list
+    building_id = system[:Location].building_id.get
+    system["StaffAPI"].systems_in_building(building_id).get.as_h.transform_values(&.as_a.map(&.as_s))
+  rescue error
+    logger.warn(exception: error) { "unable to obtain list of systems in the building" }
+    nil
+  end
+
+  # Lists external guests based on email domains
+  def list_external_guests
+    meetings = [] of Hash(String, JSON::Any)
+
+    systems.each do |level_id, system_ids|
+      system_ids.each do |system_id|
+        period_start = Time.local.to_unix
+        period_end = Time.local.to_unix + 86400
+
+        events = staff_api.query_events(period_start, period_end, systems: [system_id]).get
+
+        events.each do |event|
+          domain = event.as_h.["host"].as_s.split("@").last
```

Comment:
The host field here will be the same as the calendar/room email or the users whos calendar the event is in.
So, you should check attendees field on the event for the emails instead.

this is what the attendees field looks like.
the test@example.com attendee is external.
```
[
  {
    "name": "test@example.com",
    "email": "test@example.com",
    "response_status": "needsAction",
    "resource": false,
    "checked_in": false,
    "visit_expected": true,
    "extension_data": {}
  },
  {
    "name": "Test Room 4",
    "email": "testroom4@0cbfs.onmicrosoft.com",
    "response_status": "needsAction",
    "resource": true,
    "extension_data": {}
  },
  {
    "name": "Grady Archie",
    "email": "gradya@0cbfs.onmicrosoft.com",
    "response_status": "accepted",
    "resource": false,
    "checked_in": false,
    "visit_expected": true,
    "extension_data": {}
  }
]
```